lib/jnpr/junos/utils/sw.py

```diff
@@ -270,44 +270,82 @@ def _issu_requirement_validation(self):
                      (self._dev.facts['version_RE0'],
                       self._dev.facts['version_RE1']))
             return False
+        if not self._issu_nssu_requirement_validation():
+            return False
+        self.log('Verify that GRES is enabled on the backup Routing Engine\n'
+                 'by using the command "show system switchover"')
+        output = ''
+        try:
+            op = self._dev.rpc.request_shell_execute(routing_engine='backup',
+                                                 command="cli show system switchover")
+            output = op.findtext('.//output', default='')
+        except RpcError:
+            # request-shell-execute rpc is not available for <14.1
+            with StartShell(self._dev) as ss:
+                ss.run('cli', '> ', timeout=5)
+                if ss.run('request routing-engine login other-routing-engine')[0]:
+                    # depending on user permission, prompt will go to either cli
+                    # or shell, below line of code prompt will finally end up in
+                    # cli mode
+                    ss.run('cli', '> ', timeout=5)
+                    data = ss.run('show system switchover', '> ', timeout=5)
+                    output = data[1]
+                    ss.run('exit')
```

Comment:
Does this cleanly exit the `request routing-engine login` in all cases (both shell user and cli user)?